src/main/java/org/terasology/biomesAPI/BiomeRegistry.java

```diff
@@ -96,4 +98,21 @@
      * @return Collection of biomes of given subtype
      */
     <T extends Biome> Collection<T> getRegisteredBiomes(Class<T> biomeClass);
+
+    /**
+     * Returns all biomes that do not forbid themselves from the given location.
+     * @param region The world region
```

Comment:
The reason I created the implementation inside the biome instead of inside the registry is that it is a much simpler matter to add abstract conditional logic as an overridden method of the biome than it is to insert such logic directly into a map.

Honestly, it leaves me wondering if biomes should not be changed to a form of asset that can be handled by the entity system, like prefabs... but I believe that doing so would likely require an overhaul of the game loading system so that the entity system is initialized before facets are created.

Reply:
Well, the facet providers already use _components_ to hold their configuration (not that this is working for actual world gen, though). Turning biomes into assets sounds like a good idea, overall...